### ISSUE TITLE
Request: Option to list only the latest specified OS to use in scripting.

### ISSUE THREAD

**Issue by PicoMitchell** [ninxsoft/Mist#24]
I'm imagining using Mist in a wrapper script to keep all the macOS installers I need up-to-date at once.

In this kind of scenario, I would like to be able to know what the latest version of Big Sur or Monterey is before attempting a download so that I can check if it's already downloaded or delete the old installer if a new download is needed.

I know this data could be retrieved with `mist list -e` but it seems like overkill for my own script to have to analyze that output when Mist already knows how to do the "find the latest Big Sur" magic like it does when starting a download.

I'm imagining a command like `mist list --latest 'Big Sur'` that would JUST output the default name that would be used for an output file, or a specified naming pattern like can be done when downloading. This kind of output would also need to not include all the progress display for easy usage in a script, or maybe that could be omitted only with a `-q, --quiet` type argument. This could also be thought of as a "dry run" type action for Mist.

I'm imaging being able to write code like this (this is just quick psuedo-code):

```
if [[ -f "/Shared/Mist/$(mist list --latest 'Big Sur').dmg" ]]; then
    echo 'Big Sur Installer is up-to-date!'
else
    echo 'Big Sur Installer needs update...'
    rm -f '/Shared/Mist/Install macOS Big Sur'*'.dmg' # Delete any outdated installer.
    mist download --image 'Big Sur'
fi
```

Another related feature that would be nice is for Mist to check if a duplicate filename already exists in the download location before starting a new download (currently Mist just starts downloading and then overwrites an existing file without any warning). I'm imagining Mist could prompt whether or not to continue if a duplicate file already exists, or could just exit with a failure before downloading unless a `-f, --force` type argument is used. These changes would be great either way, but they would not be a substitute for retrieving the latest version before downloading since I would want a chance to delete any old installers before starting a new download.

Thank you so much for Mist!

**Comment by PicoMitchell** [comment]
I was able to achieve the same thing by `awk`ing the `mist list` output, but I still think this would be a useful feature :-)

```
installer_dmgs_path="${HOME}/Documents/Programming/Free Geek/MacLand Images/macOS Installers"

installer_names_to_download=( 'High Sierra' 'Catalina' 'Big Sur' 'Monterey beta' )
mist_list_output="$(mist list)"

for this_installer_name_to_download in "${installer_names_to_download[@]}"; do
    this_installer_dmg_name="Install macOS ${this_installer_name_to_download} $(echo "${mist_list_output}" | awk '/'"${this_installer_name_to_download}"'/ { if ($5 == "│") { print $6"-"$8 } else { print $7"-"$9 } exit }').dmg"

    if [[ -f "${installer_dmgs_path}/${this_installer_dmg_name}" ]]; then
        echo "\"${this_installer_dmg_name}\" is up-to-date!"
    else
        echo "\"${this_installer_dmg_name}\" needs to be downloaded..."
        rm -f "${installer_dmgs_path}/Install macOS ${this_installer_name_to_download} "*'.dmg' # Delete any outdated installer dmgs.
        sudo mist download --image "${this_installer_name_to_download}"
    fi
done
```

**Comment by PicoMitchell** [comment]
I really like the `mist list "Big Sur" --latest --quiet --output-format json` idea! That output would be very easy to `awk`, etc!

Using `--export` definitely works as well, but feels "heavy" for this kind of task. When wanting to deal with the output immediately in a script I would always prefer having `stdout` to pipe or set to a variable.

PS. I will make a new issue in the morning for the overwrite/force discussion.
